tests/test_web_response.py

```diff
@@ -257,6 +258,82 @@ def test_last_modified_reset() -> None:
     assert resp.last_modified is None
 
 
+def test_etag_initial() -> None:
+    resp = StreamResponse()
+    assert resp.etag is None
+
+
+def test_etag_string() -> None:
+    resp = StreamResponse()
+    value = "0123-kotik"
+    resp.etag = value
+    assert resp.etag == ETag(value=value)
+    assert resp.headers[hdrs.ETAG] == f'"{value}"'
+
+
+@pytest.mark.parametrize(
+    "etag,expected_header",
+    (
+        (ETag(value="0123-weak-kotik", is_weak=True), 'W/"0123-weak-kotik"'),
+        (ETag(value="0123-strong-kotik", is_weak=False), '"0123-strong-kotik"'),
+    ),
+)
+def test_etag_class(etag, expected_header) -> None:
+    resp = StreamResponse()
+    resp.etag = etag
+    assert resp.etag == etag
+    assert resp.headers[hdrs.ETAG] == expected_header
+
+
+def test_etag_any() -> None:
+    resp = StreamResponse()
+    resp.etag = "*"
+    assert resp.etag == ETag(value="*")
+    assert resp.headers[hdrs.ETAG] == "*"
+
+
+@pytest.mark.parametrize(
+    "invalid_value",
+    (
+        '"invalid"',
+        "повинен бути ascii",
+        ETag(value='"invalid"', is_weak=True),
+        ETag(value="bad ©®"),
+    ),
+)
+def test_etag_invalid_value_set(invalid_value) -> None:
+    resp = StreamResponse()
+    with pytest.raises(ValueError):
```

Comment:
The best practice is to always include `, match=r'...'` argument demonstrating the intended exception reason more precisely.